File: app/src/main/java/com/snazhmudinov/movies/models/Result.java
package com.snazhmudinov.movies.models;

import com.google.gson.annotations.SerializedName;
import com.snazhmudinov.movies.constans.Constants;

import org.parceler.Parcel;

/**
 * Created by snazhmudinov on 6/10/17.
 */
@Parcel
public class Result {
    @SerializedName("id")
    public String id;
    @SerializedName("iso_639_1")
    public String iso6391;
    @SerializedName("iso_3166_1")
    public String iso31661;
    @SerializedName("key")
    public String key;
    @SerializedName("name")
    public String name;
    @SerializedName("site")
    public String site;
    @SerializedName("size")
    public Integer size;
    @SerializedName("type")
    public String type;

    public Result() {}

    public String getId() {
        return id;
    }

    public String getIso6391() {
        return iso6391;
    }

    public String getIso31661() {
        return iso31661;
    }

    public String getKey() {
        return Constants.YOUTUBE_BASE_URL + key;
    }

    public String getName() {
        return name;
    }

    public String getSite() {
        return site;
    }

    public Integer getSize() {
        return size;
    }

    public String getType() {
        return type;
    }

}

File: app/src/main/java/com/snazhmudinov/movies/models/Result.kt
package com.snazhmudinov.movies.models

import com.google.gson.annotations.SerializedName
import com.snazhmudinov.movies.constans.Constants

import org.parceler.Parcel

/**
 * Created by snazhmudinov on 6/10/17.
 */
@Parcel
class Result {
    @SerializedName("id")
    var id: String? = null
    @SerializedName("iso_639_1")
    var iso6391: String? = null
    @SerializedName("iso_3166_1")
    var iso31661: String? = null
    @SerializedName("key")
    var key: String? = null
    @SerializedName("name")
    var name: String? = null
    @SerializedName("site")
    var site: String? = null
    @SerializedName("size")
    var size: Int? = null
    @SerializedName("type")
    var type: String? = null

    val trailerURL: String
        get() = Constants.YOUTUBE_BASE_URL + key!!

}
